src/java/org/apache/cassandra/cql3/functions/types/TypeCodec.java

```diff
@@ -3071,8 +3072,8 @@ public Duration deserialize(ByteBuffer bytes, ProtocolVersion protocolVersion)
                 DataInput in = ByteStreams.newDataInput(Bytes.getArray(bytes));
                 try
                 {
-                    int months = (int) VIntCoding.readVInt(in);
-                    int days = (int) VIntCoding.readVInt(in);
+                    int months = checkedCast(VIntCoding.readVInt(in));
+                    int days = checkedCast(VIntCoding.readVInt(in));
```

Comment:
should use the new `checked` methods rather than doing `checkedCast` here

Reply:
It's updated now.